core/arch/arm/plat-k3/drivers/ti_sci.c

```diff
@@ -68,11 +71,16 @@ static int ti_sci_setup_xfer(uint16_t msg_type, uint32_t msg_flags,
 	}
 
 	hdr = (struct ti_sci_msg_hdr *)tx_buf;
+
+	exceptions = cpu_spin_lock_xsave(&ti_sci_spin_lock);
+
 	hdr->seq = ++message_sequence;
 	hdr->type = msg_type;
 	hdr->host = OPTEE_HOST_ID;
 	hdr->flags = msg_flags | TI_SCI_FLAG_REQ_ACK_ON_PROCESSED;
 
+	cpu_spin_unlock_xrestore(&ti_sci_spin_lock, exceptions);
```

Comment:
I think only `message_sequence` incrementing needs to be protected.

Reply:
Done.